api/controllers/user-action.js

```diff
@@ -0,0 +1,36 @@
+const { showActions } = require('../authorizers/site');
+const { UserAction, User, ActionType } = require('../models');
+const userActionSerializer = require('../serializers/user-action');
+
+const validId = (maybeId) => {
+  const idToNumber = Number(maybeId);
+
+  return Promise.resolve(idToNumber)
+  .then((id) => {
+    if (isNaN(id)) {
+      throw 404;
+    }
+
+    return id;
+  });
+};
+
+module.exports = {
+  find(req, res) {
+    validId(req.params.site_id)
+    .then(id => showActions(req.user, { id }))
+    .then(siteId =>
+      UserAction.findAllBySite({ UserAction, User, ActionType }, siteId)
```

Comment:
Perhaps a new "service" would work well, maybe in a file called `UserActionFinder.js`:

```js
const { UserAction, User, ActionType } = require('../models');

const findAllBySite = (siteId) =>
  UserAction.findAll({
    where: {
      siteId,
    },
    attributes: ['id', 'targetType', 'siteId', 'createdAt'],
    include: [{
      model: User,
      as: 'actionTarget',
      attributes: ['id', 'username', 'email', 'createdAt'],
    },
    {
      model: ActionType,
      as: 'actionType',
      attributes: ['action'],
    }],
  });
```

Reply:
Or call it `services/UserAction.js` and move both this `findAllBySite` method and the existing `services/UserActionCreator.js`'s `buildUserAction` method into it. I _think_ dealing with the model dependencies is part of the reasoning we have those service modules (though they do predate me, so I'm not 100% sure).

Reply:
So I made this choice for a couple of reasons:

1). There isn't a way, that I could find, to arbitrarily define new methods on a model. In Rails, you could just write a new finder method and be able to access the actual static class. In sequelize, you have to pass the actual class back into itself when you define static methods — this is what we do in the `associate` methods, for example. 

Initially I tried to use a named scope, but includes don't seem to work properly there.

2). To me a service is useful for things like combining behaviors form multiple classes, proxying/wrapping an existing external API, etc.. The method here is basically a SQL statement written using sequelize's helpers. To me they aren't dependencies per se, but I don't know exactly how to define dependencies in the context of database design.

That was my line of thinking anyway, let me know if that's lame 😄

Reply:
I suppose one way around this is to define a helper get included in the class file, and put the exposed method from that helper into the `classMethods` object in the model definition.

Reply:
Sequelize is so clunky :\ This seems like it should be a simple method to add, but of course it isn't

I want to think through it a little more. It would be nice to have a general approach we can use for methods like this. I definitely don't like passing models to functions like this.

Reply:
I haven't thought through it any more, but what about inverting the call, so it is on the `Site` model? Like `Site.getUserActions()`?

Reply:
I initially had it there, it just seemed more like it belonged on the UserAction model. The thing is, we still have to pass the `Site` to itself, since this find needs to be done against the class, not the instance.